### ISSUE TITLE
Replace Reachability framework with one that correctly determines network status

### ISSUE THREAD

**Issue by einsteinx2** [balancemymoney/balance-open#99]
Right now if you have a wifi connection but now internet access, it will act as if there is a connection and start syncing.

**Comment by einsteinx2** [comment]
So interestingly, this is actually "working" as intended. From Apple's docs:

> Note: Reachability cannot tell your application if you can connect to a particular host, only that an interface is available that might allow a connection, and whether that interface is the WWAN.

So that doesn't really help us. Looks like we need to manually check host availability after Reachability says we have a connection.

**Comment by einsteinx2** [comment]
Found a reasonable solution. I added a hello endpoint on our new backend that just returns the word "hello". When the reachability class determines the network is available, we poll that address immediately to see if we get the correct response. If it succeeds, we send the network available notification to begin syncing. If it fails, we retry every second until it either succeeds or the reachability class determines the network is unavailable.

This should fix the issue whereby the network interface is ready, but network connectivity isn't, causing initial syncs to fail.

Also I moved the Reachability.swift file into our project instead of using it as a Carthage framework since it's literally 1 file.
